portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,37 +14,86 @@
-    String[] words = text.split("\\s*,\\s*");
-
-    // Respond with the result.
-    response.setContentType("text/html;");
-    response.getWriter().print("A message from: ");
-    response.getWriter().println(Arrays.toString(names));
-    response.getWriter().println(Arrays.toString(words));
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    // Get integer input for limited number of comments.
+    int limit;
+    try {
+      limit = Integer.parseInt(getParameter(request, "limit", ""));
+    }
+    catch (NumberFormatException e) {
+      limit = 0;
+    }
+    // Private class which is supposed to act like a Comment class.
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable(FetchOptions.Builder.withLimit(limit))) {
+      long id = entity.getKey().getId();
+      String title = (String) entity.getProperty("title");
+      long timestamp = (long) entity.getProperty("timestamp");
+     
+      Comment comment = new Comment(id, title, timestamp);
+      comments.add(comment);
+    }
+    Gson gson = new Gson();
+
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(comments));
   }
 
-  /**
-   * @return the request parameter, or the default value if the parameter
-   *         was not specified by the client
-   */
+  /* Do Post function responsible for creating new comments. */ 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      // Method to create and send New Comment to the server.
+      String title = request.getParameter("title");
+      long timestamp = System.currentTimeMillis();
+
+      Entity commentEntity = new Entity("Comment");
+      commentEntity.setProperty("title", title);
+      commentEntity.setProperty("timestamp", timestamp);
+      
+      // Create a server connection to get data and put new comments to teh database.
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(commentEntity);
+
+      // After the procedure go back to the home page.
+      response.sendRedirect("/index.html");
+  }
+   /**
+    * @param request - Defines an object to provide client request information to a servlet.
+    * @param name - Name of the requested element (can be name of teh text-box in teh form).
+    * @param defaultValue - The value, related to the element, that was requested (could be true/false).
+    * @return - The request parameter, or the default value if the parameter
+   *         was not specified by the client.
+    */
+   /* Private function accessing the properties of the Comment-objects. */
```

Comment:
usually you'd put the function description at the start of the javadoc comment, rather than having a seperate comment after